cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/nosql/NoSqlStructuredTable.java

```diff
@@ -490,6 +516,151 @@ private void closeScanner() {
     }
   }
 
+  /**
+   * Filters elements matching an index  {@link ScannerIterator}.
+   */
+  @VisibleForTesting
+  static final class FilterByIndexIterator extends AbstractCloseableIterator<StructuredRow> {
+    private final Iterator<? extends CloseableIterator<StructuredRow>> scannerIterator;
+    private CloseableIterator<StructuredRow> currentScanner;
+    private final Field<?> filterIndex;
+    private final StructuredTableSchema schema;
+
+    FilterByIndexIterator(Iterator<? extends CloseableIterator<StructuredRow>> scannerIterator, Field<?> filterIndex,
```

Comment:
this this ever called with more than one scanner iterator? If not it's simpler just to take a single iterator

Reply:
Agree here that we perhaps don't need the iterator of iterator here

Reply:
I was thinking of maintaining consistency with the other iterators - `LimitIterator`, but yes, we don't need more than one